eslint.config.mjs

```diff
@@ -1,6 +1,62 @@
 import baseConfig from '@hono/eslint-config'
 import { defineConfig, globalIgnores } from 'eslint/config'
 
-export default defineConfig(globalIgnores(['.yarn', '**/dist']), {
+export default defineConfig(globalIgnores(['.yarn', '**/coverage', '**/dist']), {
   extends: baseConfig,
+
+  languageOptions: {
+    parserOptions: {
+      projectService: true,
+      tsconfigRootDir: import.meta.dirname,
+    },
+  },
+
+  linterOptions: {
+    reportUnusedDisableDirectives: 'error',
+    reportUnusedInlineConfigs: 'error',
+  },
+
+  rules: {
+    '@typescript-eslint/array-type': 'off',
+    '@typescript-eslint/await-thenable': 'off',
+    '@typescript-eslint/consistent-indexed-object-style': 'off',
+    '@typescript-eslint/consistent-type-definitions': 'off',
+    '@typescript-eslint/dot-notation': 'off',
+    '@typescript-eslint/no-base-to-string': 'off',
+    '@typescript-eslint/no-confusing-void-expression': 'off',
+    '@typescript-eslint/no-deprecated': 'off',
+    '@typescript-eslint/no-duplicate-type-constituents': 'off',
+    '@typescript-eslint/no-dynamic-delete': 'off',
+    '@typescript-eslint/no-floating-promises': 'off',
+    '@typescript-eslint/no-invalid-void-type': 'off',
+    '@typescript-eslint/no-misused-promises': 'off',
+    '@typescript-eslint/no-non-null-assertion': 'off',
+    '@typescript-eslint/no-redundant-type-constituents': 'off',
+    '@typescript-eslint/no-unnecessary-boolean-literal-compare': 'off',
+    '@typescript-eslint/no-unnecessary-condition': 'off',
+    '@typescript-eslint/no-unnecessary-template-expression': 'off',
+    '@typescript-eslint/no-unnecessary-type-arguments': 'off',
+    '@typescript-eslint/no-unnecessary-type-assertion': 'off',
+    '@typescript-eslint/no-unnecessary-type-parameters': 'off',
+    '@typescript-eslint/no-unsafe-argument': 'off',
+    '@typescript-eslint/no-unsafe-assignment': 'off',
+    '@typescript-eslint/no-unsafe-call': 'off',
+    '@typescript-eslint/no-unsafe-enum-comparison': 'off',
+    '@typescript-eslint/no-unsafe-member-access': 'off',
+    '@typescript-eslint/no-unsafe-return': 'off',
+    '@typescript-eslint/no-useless-constructor': 'off',
+    '@typescript-eslint/non-nullable-type-assertion-style': 'off',
+    '@typescript-eslint/only-throw-error': 'off',
+    '@typescript-eslint/prefer-function-type': 'off',
+    '@typescript-eslint/prefer-includes': 'off',
+    '@typescript-eslint/prefer-nullish-coalescing': 'off',
+    '@typescript-eslint/prefer-optional-chain': 'off',
+    '@typescript-eslint/prefer-regexp-exec': 'off',
+    '@typescript-eslint/prefer-return-this-type': 'off',
+    '@typescript-eslint/require-await': 'off',
+    '@typescript-eslint/restrict-plus-operands': 'off',
+    '@typescript-eslint/restrict-template-expressions': 'off',
+    '@typescript-eslint/unbound-method': 'off',
+    '@typescript-eslint/unified-signatures': 'off',
+  },
```

Comment:
I've initially disabled all the new rules to keep this PR small

I'd be happy to revisit each rule individually and decide if it's worth enabling or not. If a new rule is problematic, it can always be disabled in the shared eslint config

My personal preference is to have less configuration, and rely on tooling to make these decisions for me. But I understand that doesn't fit with how everyone works 😅

Reply:
> I've initially disabled all the new rules to keep this PR small

I like this approach. Fixing lint errors is troublesome, and the diffs will be huge. We don't have to do it.